src/main/scala/com/mongodb/spark/config/WriteConfig.scala

```diff
@@ -41,24 +43,66 @@ object WriteConfig extends MongoOutputConfig {
    * @return the write config
    */
   def apply(databaseName: String, collectionName: String, writeConcern: WriteConcern): WriteConfig =
-    new WriteConfig(databaseName, collectionName, WriteConcernConfig(writeConcern))
+    new WriteConfig(databaseName, collectionName, None, WriteConcernConfig(writeConcern))
+
+  /**
+   * Creates a WriteConfig
+   *
+   * @param databaseName the database name
+   * @param collectionName the collection name
+   * @param connectionString the optional connection string used in the creation of this configuration
+   * @param writeConcern the WriteConcern to use
+   * @return the write config
+   */
+  def apply(databaseName: String, collectionName: String, connectionString: String, writeConcern: WriteConcern): WriteConfig =
```

Comment:
It may make sense to include localThreshold in WriteConfig as well.  It's meaningless for a standalone or a replica set, but it allows you to have some control over mongoS targeting of writes.